pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -301,9 +301,9 @@ def get_dfds_dcds(model, theta_names, tee=False, solver_options=None):
     gradient_c: scipy.sparse.csr.csr_matrix
         Ncon by Nvar size sparse matrix. A Jacobian matrix of the
         constraints with respect to the (decision variables, parameters)
-        at the optimal solution. Each row contains [column number,
-        row number, and value], column order follows variable order in col
-        and index starts from 1. Note that it follows k_aug.
+        at the optimal solution. Each row contains [row number,
+        column number, and value], column order follows variable order in col
+        and index starts from 0. Note that it follows k_aug.
         If no constraint exists, return []
```

Comment:
Is this docstring describing the structure of a CSR matrix? What does "row" refer to for a CSR matrix? The sub-arrays in the `indices` and `data` arrays corresponding to a particular row? Unless this is documenting something I'm missing, I would just cut out everything after `at the optimal solution.` Also, if no constraints exist, this docstring is incorrect, because we return an empty array rather than an empty list. In this case we should probably just return an empty CSR matrix...

Reply:
I think this is describing the structure of a CSR matrix. @jialuw96 take a look at this: https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.html

I agree with the suggestion to return an empty CSR matrix.

Reply:
The docstring is saying (I think) that the rows and columns of gradient_c correspond to the equations and variables, respectively. @Robbybp Do you agree with this? I agree the description of the CSR matrix is confusing. I'd say just remove the [ ] and reword the sentence after "optimal".

Reply:
Yes, I agree that rows correspond to equations and columns correspond to variables. It is a little confusing because k_aug appears to print out the (row, col, data) of the transpose of this matrix, then this method switches the row and col entries to return this gradient_c, which I think is the more intuitive representation.

Reply:
@jialuw96 We should update the docstring (or comments or both) to reflect this nuance.